api/transport/response.go

```diff
@@ -20,13 +20,17 @@
 
 package transport
 
-import "io"
+import (
+	"go.uber.org/yarpc/yarpcerrors"
+	"io"
+)
 
 // Response is the low level response representation.
 type Response struct {
-	Headers          Headers
-	Body             io.ReadCloser
-	ApplicationError bool
+	Headers            Headers
+	Body               io.ReadCloser
+	ApplicationError   bool
+	ApplicationFailure *ApplicationFailure
```

Comment:
Reading this, it took me some time to understand the difference between `ApplicationError` and `ApplicationFailure`. The terms "error" and "failure" are basically synonyms in this context. Perhaps a clearer name could be `ApplicationErrorDetails` or something similar?

Reply:
Sounds good! Will rename it to `ApplicationErrorMeta`.